### PR TITLE
Dynamically generate the applications links on the logged in page

### PR DESCRIPTION
**Pull request by lgarvey**

### CODE REVIEW

File: sso/user/models.py [hunk 1]
```diff
@@ -264,19 +264,30 @@ def get_application_username(self, application=None):
         else:
             return self.get_emails_for_application(application)[0]
 
-    def get_permitted_applications(self):
+    def get_permitted_applications(self, public_only=False):
```
Comment (michalc) [inline sso/user/models.py]: It's a borderline style thing, but should this be a "positive permission" type variable, e.g. `include_private` or `include_non_public` / similar?

I might even go as far as to change existing call sites to make it explicit that non public apps are included there, and not allowing a default at all.

File: sso/user/models.py [hunk 2]
```diff
@@ -264,19 +264,30 @@ def get_application_username(self, application=None):
...
+
+        if public_only:
+            apps = {ap for ap in apps if ap.public}
+
+        results = [
+            {
+                'key': app.application_key,
+                'url': app.start_url,
+                'name': app.display_name
+            } for app in apps
```
Comment (michalc) [inline sso/user/models.py]: Very minor style-type thing, so all good to ignore... but you could have an `if` filter here and avoid the `if public_only` block above.

Comment (lgarvey) [reply sso/user/models.py]: Do you mean an `if expression`?  To avoid mutating the var?

File: sso/user/models.py (hunk 1 above)
Comment (lgarvey) [reply sso/user/models.py]: I prefer this solution to my own 👍 I'll get it changed.

File: sso/user/models.py (hunk 2 above)
Comment (michalc) [reply sso/user/models.py]: Yes (sorry I was unclear)

**Pull request merged**